components/resources/library/src/jsMain/kotlin/org/jetbrains/compose/resources/Resource.js.kt

```diff
@@ -22,42 +8,35 @@ import org.w3c.dom.parsing.DOMParser
 @Suppress("unused")
 @ExperimentalResourceApi
 object WebResourcesConfiguration {
+    internal var resourcePathCustomization: (path: String) -> String = { "./$it" }
 
     /**
-     * An internal default factory method for creating [Resource] from a given path.
-     * It can be changed at runtime by using [setResourceFactory].
-     */
-    @ExperimentalResourceApi
-    internal var jsResourceImplFactory: (path: String) -> Resource = { urlResource("./$it") }
-
-    /**
-     * Sets a custom factory for the [resource] function to create [Resource] instances.
-     * Once set, the [factory] will effectively define the implementation of the [resource] function.
+     * Sets a customization function for resource path. This allows you to modify the resource path
+     * before it is used.
      *
-     * @param factory A lambda that accepts a path and produces a [Resource] instance.
-     * @see configureWebResources for examples on how to use this function.
+     * @param customization the customization function that takes a path String and returns a modified path String
      */
     @ExperimentalResourceApi
-    fun setResourceFactory(factory: (path: String) -> Resource) {
-        jsResourceImplFactory = factory
+    fun setResourcePathCustomization(customization: (path: String) -> String) {
+        resourcePathCustomization = customization
     }
 }
 
 /**
  * Configures the web resources behavior.
  *
- * Allows users to override default behavior and provide custom logic for generating [Resource] instances.
+ * Allows users to override default behavior and provide custom logic for generating resource's paths.
  *
  * @param configure Configuration lambda applied to [WebResourcesConfiguration].
  * @see WebResourcesConfiguration For detailed configuration options.
  *
  * Examples:
  * ```
  *  configureWebResources {
- *     setResourceFactory { path -> urlResource("/myApp1/resources/$path") }
+ *     setResourceFactory { path -> "/myApp1/resources/$path" }
```

Comment:
It is renamed to setResourcePathCustomization

Reply:
BTW, I am not sure about naming. What about setResourcePathMapping?

Reply:
Since it is a configuration DSL I changed it to
```kotlin
fun main() {
    configureWebResources {
        resourcePathMapping { path -> "./$path" }
    }
    onWasmReady {
        CanvasBasedWindow("Resources demo") {
            UseResources()
        }
    }
}
```